src/ir/ty.rs

```diff
@@ -538,32 +538,10 @@ impl TemplateParameters for TypeKind {
 }
 
 impl CanDeriveDebug for Type {
-    type Extra = ();
+    type Extra = Item;
 
-    fn can_derive_debug(&self, ctx: &BindgenContext, _: ()) -> bool {
-        match self.kind {
-            TypeKind::Array(t, len) => {
-                len <= RUST_DERIVE_IN_ARRAY_LIMIT && t.can_derive_debug(ctx, ())
-            }
-            TypeKind::ResolvedTypeRef(t) |
-            TypeKind::TemplateAlias(t, _) |
-            TypeKind::Alias(t) => t.can_derive_debug(ctx, ()),
-            TypeKind::Comp(ref info) => {
-                info.can_derive_debug(ctx, self.layout(ctx))
-            }
-            TypeKind::Pointer(inner) => {
-                let inner = ctx.resolve_type(inner);
-                if let TypeKind::Function(ref sig) =
-                    *inner.canonical_type(ctx).kind() {
-                    return sig.can_derive_debug(ctx, ());
-                }
-                return true;
-            }
-            TypeKind::TemplateInstantiation(ref inst) => {
-                inst.can_derive_debug(ctx, self.layout(ctx))
-            }
-            _ => true,
-        }
```

Comment:
Are there any other `CanDeriveDebug` impls that are now unused and we can remove? I'd try commenting all of them out other than `Item` and `ItemId` and see if any call sites start failing to compile, and then look at each of those call sites and determine whether it would be easier to just use an `Item`/`ItemId` impl or to uncomment the impl that was being used. I expect most of them are handled by `Item` and `ItemId`.

We should at minimum be able to delete the `detect_can_derive_debug_cycle` member from `CompInfo` and its `CanDeriveDebug` impl as well.